packages/fxa-content-server/app/scripts/views/settings/change_password.jsx

```diff
@@ -0,0 +1,319 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import AuthErrors from '../../lib/auth-errors';
+import BackMixin from '../mixins/back-mixin';
+import Cocktail from 'cocktail';
+import FormView from '../form';
+import ExperimentMixin from '../mixins/experiment-mixin';
+import PasswordMixin from '../mixins/password-mixin';
+import PasswordStrengthMixin from '../mixins/password-strength-mixin';
+import ServiceMixin from '../mixins/service-mixin';
+import SettingsPanelMixin from '../mixins/settings-panel-mixin';
+// import Template from 'templates/settings/change_password.mustache';
+// eslint-disable-next-line
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Translator from '../../lib/translator';
+
+const translator = new Translator();
+
+const t = msg => translator.get(msg);
+
+function ChangePassword(props) {
+ return (
+  <div id="change-password" className="settings-unit">
+   <ChangePasswordHeader />
+   <ChangePasswordForm
+      account={props.account}
+      submit={props.submit}
+      displayError={props.displayError}
+      isPanelOpen={props.isPanelOpen}
+      showValidationError={props.showValidationError}
+   />
+  </div>
+ );
+}
+
+export function ChangePasswordHeader() {
+ return (
+  <div className="settings-unit-stub">
+   <header className="settings-unit-summary">
+    <h2 className="settings-unit-title">{t('Password')}</h2>
+   </header>
+   <button
+    className="settings-button secondary-button settings-unit-toggle"
+    data-href="settings/change_password"
+   >
+    {t('Change...')}
+   </button>
+  </div>
+ );
+}
+
+export class ChangePasswordForm extends React.Component {
+ constructor(props) {
+  super(props);
+
+  this.state = {
+   hasFocused: false,
+   mail: props.account.get('email') || '',
+   newPass: '',
+   newVPass: '',
+   oldPass: '',
+  };
+ }
+ setOldPassword = event => {
+  this.setState({
+   oldPass: event.target.value,
+  });
+ };
+
+ setNewPassword = event => {
+  this.setState(
+   {
+    newPass: event.target.value,
+   });
+ };
+
+ setNewVPassword = event => {
+  this.setState(
+   {
+    newVPass: event.target.value,
+   });
+ };
+ showValidationErrorsEnd() {
+  if (this.state.newPass && this.validateFormField('#new_password') && this.state.newVPass && this.state.newPass !== this.state.newVPass) {
+   const err = AuthErrors.toError('PASSWORDS_DO_NOT_MATCH');
+   this.props.showValidationError('#new_vpassword', err);
+  }
+ }
+ componentDidUpdate() {
+  if (!this.state.hasFocused && this.props.isPanelOpen) {
+   this._input.focus();
+   this.setState({ hasFocused: true });
+  }
+  this.showValidationErrorsEnd();
+ }
+ componentWillUnmount() {
+  return this.state.newPass === this.state.newVPass;
+ }
+
+ setFocus = () => {
+  this.setState(
+   {
+    newPass: '',
+    newVPass: '',
+    oldPass: '',
+    hasFocused: false,
+   },
+   () => {
+    this.componentDidUpdate();
```

Comment:
Will `componentDidUpdate` be called automatically due to the `setState`?

Reply:
Ya it would be called due to `setState` but because of that the `hasFocused` property is being set to True. So we are calling the `componentDidUpdate` to restore its value to false.